test/e2e/json-rpc/wallet_revokePermissions.spec.js

```diff
@@ -0,0 +1,61 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  unlockWallet,
+  openDapp,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Revoke Dapp Permissions', function () {
+  it('should revoke dapp permissions ', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        await openDapp(driver);
+
+        let accountsDiv = await driver.findElement('#accounts');
+
+        assert.equal(
+          await accountsDiv.getText(),
+          '0x5cfe73b6021e818b776b421b1c4db2474086a7e1',
+        );
+
+        // wallet_revokePermissions request
+        const revokePermissionsRequest = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'wallet_revokePermissions',
+          params: [
+            {
+              eth_accounts: {},
+            },
+          ],
+        });
+
+        const result = await driver.executeScript(
+          `return window.ethereum.request(${revokePermissionsRequest})`,
+        );
+
+        // Response of method call
+        assert.deepEqual(result, null);
+
+        // Reload Dapp
+        await driver.executeScript(`window.location.reload()`);
```

Comment:
Had to force reload the dapp for the accounts label to reset, but the normal behavior is that the label automatically clears with the permission revoked.

Reply:
> but the normal behavior is that the label automatically clears with the permission revoked.

are you saying its not auto clearing in the test but does so when manually testing?

Reply:
is this a test dapp bug then?